docs/agent-reference/agent-core/configuration-manager.md

```diff
@@ -0,0 +1,95 @@
+# Device Update Agent Configuration
+
+## APIs
+
+```c
+// A config object handle.
+typedef void* CONFIG_HANDLE
+
+// Instancitate a config object from specified file.
+CONFIG_HANDLE LoadConfig(const char * configFilePath);
+
+// Free all resources allocated for the config object.
+void FreeConfig(CONFIG_HANDLE config);
+
+// Returns value of the specified config name.
+// (Output is a json string).
+const char* GetValue(const char* name);
+
+```
+
+## Configuration File Format
+
+See example `du-config.json` file format below:
+
+```json
+
+"enabledIoTHubTracing" : true,
+"logLevel" : 0,
+
+"aduShellTrustedUsers": ["adu", "do"],
+
+"agents":[
+  {
+      "name":"host-update",
+      "runas":"adu",
+      "connectionInfoSource":"AIS",
+      "connectionData":{
+          "connectionType" : "module",
+          "moduleName" : "ioTHubDeviceUpdate"
+      },
+
+      "aduc_manufacturer" : "Contoso",
+      "aduc_model" : "Smart-Box"
+  },
+
+  {
+      "name":"leaf-update",
+      "runas":"adu",
+      "connectionInfoSource":"string",
+      "connectionData":{
+          "connectionString" : "HostName=..."
+      },
+
+      "aduc_manufacturer" : "Fabrikam",
+      "aduc_model" : "Camera"
+  }
+],
+
+"contentHandlers":[
+    {
+        "updateType":"microsoft/apt:1",
+        "path":"/user/lib/adu/content-handlers.d/microsoft-apt-1/libmicrosoft-apt-1.so",
+        "sha256":"xAbsdf802x3233="
+    },
+    {
+        "updateType":"microsoft/swupdate:1",
+        "path":"/user/lib/adu/content-handlers.d/microsoft-swupdate-1/libmicrosoft-swupdate-1.so",
+        "sha256":"xAbsdfsd23rs302x3123="
+    },
+    {
+        "updateType":"microsoft/mcu:1",
+        "path":"/user/lib/adu/content-handlers.d/microsoft-mcu-1/libmicrosoft-mcu-1.so",
+        "sha256":"xAbsdf802x3233="
+    },
+]
+```
+
+Property Name | Type | Description
+|----|----|----|
+|enabledIoTHubTracing|boolean|Enable IoT Hub SDK tracing<br>Default is 'false'|
+|logLevel|number|Set log verbosity<br>0 - debug<br>1 - info<br>2 - warning<br>3 - error|
+|aduShellTrustedUsers|Array of string|List of users that allowed to launch adu-shell|
+|agents|Array of [Agent Info](#agent-info)| List of child agent's information|
```

Comment:
>agents|Array of [Agent Info](#agent-info)| List of child agent's information| [](start = 1, length = 77)

Should we have a seperate value for the host? Or will it be understood that this list of DU agent's will always have the first one as the host? In other words we have a requirement that there is at least (1) agent in this field otherwise we err out.

Reply:
Currently, if no targetNames, targetGroups, nor targetClasses is specified, the update will be applied to host.